testhelper/convenience.go

```diff
@@ -167,15 +167,13 @@ func deepDiffEqual(expected, actual reflect.Value, visited map[visit]bool, path
 
 		for _, k := range keys {
 			expectedValue := expected.MapIndex(k)
-			actualValue := expected.MapIndex(k)
+			actualValue := actual.MapIndex(k)
 
 			if !expectedValue.IsValid() {
-				logDifference(path, nil, actual.Interface())
-				return
```

Comment:
No, that won't work. A good test to look at to trigger this area of code is `gophercloud/testing/params_test.go`.